### ISSUE TITLE
"Report renderElement bug #1"

### ISSUE THREAD

**Issue by setharnold** [matterhorn-chat/matterhorn#674]
Hello, a co-worker pasted a link:

https://gitlab.com/apparmor/apparmor/-/merge_requests/649

This caused the text to be displayed `<(Report renderElement bug #1)>`.

Thanks

**Comment by jtdaugherty** [comment]
What version of Matterhorn triggered this?

**Comment by setharnold** [comment]
Since I use @popey's snap package https://github.com/popey/matterhorn-snap I know it as version:

50200.10.2-132-gbb01a7ac:

You're right, when I paste that link into a message it just turns into a link. I'll have to ask my coworker if he did anything interesting while sending the message. I only recall that it had several `@someone` mentions in the message as well.

Since this url relies on resources we don't control, it's possible, if unlikely, for the contents of the url to have changed between when my coworker sent that message, and when we're retesting it now.

Thanks

**Comment by jtdaugherty** [comment]
Okay, thanks! For what it's worth, the *contents* (i.e. the data you'd get if you fetched the URL) won't have any impact on Matterhorn because Matterhorn never fetches the contents of URLs that are embedded in messages.

**Comment by jtdaugherty** [comment]
Looking at the source, that bug exception is thrown when some assumptions don't hold in our message-rendering code. I'll take a look at how that could happen. Thanks for reporting this!

**Comment by jrjohansen** [comment]
The message was to nicks followed by a colon and then a url. I will note that the nicks did not have `@` prepended. The message in its entirety was

`emitorino, jdstrand: https://gitlab.com/apparmor/apparmor/-/merge_requests/649`

**Comment by jtdaugherty** [comment]
@jrjohansen does that message still consistently cause crashes for you?

**Comment by jtdaugherty** [comment]
(Sorry, I didn't mean crashes; I forgot that this doesn't cause crashes, just produces the text of the message that was reported here.)

**Comment by jtdaugherty** [comment]
It would also seem that we report that text when there is a hyperlink in a message with a label. The message that @jrjohansen provided above just has a hyperlink (no label). To reproduce this I'll need the exact Markdown input text that resulted in the reported error.

**Comment by jtdaugherty** [comment]
@jrjohansen is it possible to go back to the message that still renders that way and get the original Markdown input text (by trying to edit the message)?

**Comment by jtdaugherty** [comment]
I was able to reproduce this with a doubly-embedded link:
```
[[test](https://gitlab.com/apparmor/apparmor/-/merge_requests/649)](https://gitlab.com/apparmor/apparmor/-/merge_requests/649)
```

**Comment by setharnold** [comment]
Thanks @jrjohansen -- I asked John to repeat the message, and in scrollback I got a second instance of the crash:

```
[01:57:30] jjohansen: seth-arnold: "emitorino, jdstrand: <(Report renderElement bug #1)>"
```

John, do I remember that you're using the matterircd bridge? Would that do any message re-writing?

Thanks

**Comment by jtdaugherty** [comment]
This is now fixed by a7f919e746d2245a85c60daa99a8632e5724014b and I'll be putting out a bugfix release later today. I'll update here when it's ready.

**Comment by jtdaugherty** [comment]
The release is ready:

https://github.com/matterhorn-chat/matterhorn/releases/tag/50200.10.3

Thanks for reporting this!
